### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by ariaveta** [aardgoose/CaveView.js#10]
Hello, I'd like to understand how to enable by default some features like showing terrain and map overlay shading. 

Any hint?  Any example?
Thanks

**Comment by aardgoose** [comment]
Hi,

I need to document terrain creation, do you have a survey/area in mind?

If you could provide an example survey - preferably a survex .3d model with the coordinate reference system defined using the '*CS OUT" command, I can document the process for you.   

Adding overlays is relatively simple when terrain data is available.

**Comment by ariaveta** [comment]
Hello,
I meant not creation of 3d models.
I simply need to show terrain by default in Caveview. I already have some 3d models, you can see them here:

http://www.speleotoscana.it/programmi_php_mm/caveview.html

You can enable terrain and shadowing and also a map.

But I need terrain and shadowing enabled as the user opens the page, without being forced to use settings menu. I guess it's only a matter to set some javascript variables, but I found no examples about that.

Thanks

**Comment by aardgoose** [comment]
Hi,

OK, I see what you want,  It is not possible in a reliable way at the moment, however should be very simple to add, it was something I had planned to do.

I'll sort this out.

**Comment by aardgoose** [comment]
Hi,

You can now set initial view settings with the latest version. The 1.4.0 release notes give an example of this. Examples of the properties of the view object can be seen in the script that drives the display on the project home page:

https://github.com/aardgoose/CaveView.js/blob/gh-pages/CaveView/lib/demoscript.js

**Comment by ariaveta** [comment]
Hello.
great job, thanks!

**Comment by aardgoose** [comment]
Hi,

The latest version (1.5.0) now has much better wall rendering for .lox models

**Comment by ariaveta** [comment]
Thanks, we used 1.5 on our website (unfortunately it's in Italian):

http://www.speleotoscana.it/3d/

You can explore 3D models of some caves in Tuscany.

Just a little note: if you set HUD: true in initial settings HUD don't get displayed, but the radio button is selected. If you cycle the radio button, you get HUD.

**Comment by aardgoose** [comment]
Hi,

Nice looking models.... 

I'll take a look at the HUD issue - the initial display setting needs some revision.
The next version will use much less memory.

If you are interested I will look at making it possible to add alternative languages for the menus.

**Comment by ariaveta** [comment]
Hello,
thanks for your email.

Yes, I'm interested and, of course, I can translate it in Italian.

Ciao.



2018-01-19 18:19 GMT+01:00 aardgoose <notifications@github.com>:

**Comment by ariaveta** [comment]
Hello, could you please provide a bit of documentation about new features usage?

Thanks.

I noticed map overlay in our models is not enabled by default in 1.6 but it was in 1.5. Here's the initial settings we use:


view: {
   section: 0,
   entrances: false,
   cameraType: CV.CAMERA_ORTHOGRAPHIC,
   box: false,
   terrain: true,
   view: CV.VIEW_ELEVATION_E,
   shadingMode: CV.MAP_OVERLAY,
   terrainShading: CV.SHADING_OVERLAY,
   walls: true,
   HUD: false,
   autoRotate: false,
   autoRotateRate: 1.0,
   stations: false,
   stationLabels: false,
   fullscreen: false
   }

**Comment by aardgoose** [comment]
Hi,  the overlay not showing is a bug I've introduced. Fix coming soon
Documentation of language etc to come (a bit busy at the moment)

**Comment by aardgoose** [comment]
In case you haven't noticed I have added some quick info on the language translation mechanism.

**Comment by ariaveta** [comment]
I admit not to have seen those info.

Where can I find them?

Thanks


2018-03-08 12:08 GMT+01:00 aardgoose <notifications@github.com>:

**Comment by aardgoose** [comment]
(linked from the demo page)

https://aardgoose.github.io/CaveView.js/translation.html

**Comment by MaticDiba** [comment]
Hi,
this really looks awesome. 

Is it possible to make detailed tutorial? 

I have terrain images (geotiff) available. If I understand correctly it is also possible to use some tile provider like OSM? 

The issue I have it, that terrain image is in different CRS. So I am a bit lost how to solve this.

I can provide terrain image and .3d file if that would help.

Othervise, this plugin really rocks :)

**Comment by aardgoose** [comment]
Hi,

Thanks, I know the documentation needs improving. If you can provide a terrain file and .3d file that would be great. I can use that as a test case to document the process of creating height tiles. I can handle CRS conversion and will document that.

For the terrain image overlays I have used Bing/OSM and some WMTS providers, again this needs documenting properly.

**Comment by MaticDiba** [comment]
Hi,
thanks for fast respond. I've uploaded a zip to [wetransfer](https://we.tl/t-IGLxpuKscb). There is survex project included with img file that represents terrain image. I'm not sure if that is what is needed for terrain generation in CaveView. Other then that I can provide .asc files for that area (one example is in ExampleOfAsci folder). 

If there is anything else that I can help with, just let me know. Also regarding helping making tutorial/example. I'm just stuck at this point of how to use terrain image that I have right now.

CRS that I use are in .svx file.

Thanks for your help.

**Comment by aardgoose** [comment]
Hi, Thanks, does the DTM use the same CRS as the Survex data?  It looks like it does.

**Comment by MaticDiba** [comment]
Yes it does. I didn't manage to get this info with dtm (I'm still learning the awesome stuff in GIS world :)).

I got .asc file from our national agency, and created .tiff with SAGA GIS, so I was able to use it in qGis.

**Comment by aardgoose** [comment]
I've made a small set of tiles to cover your example cave.

You can drag  and drop onto the demo site at https://aardgoose.github.io/CaveView.js/fulldemo.html

It looks right to me, the tiff doesn't quite cover the region but looks like a match to the images on katasterjam.si.

Just got to write up the procedure now.

I use a third GIS product to create the tiles, GRASS GIS which you probably have because it comes bundled with QGIS.

**Comment by MaticDiba** [comment]
Hi, sorry I don't quite underatan where tiles are :)

By the way ... I used CaveView on katasterjam, so users can just upload 3d to a cave or multiple caves, and then dynamically display it with CaveView: https://www.katasterjam.si/Caves/Details/10904#dokumenti

I was thinking of creating tiles for uploaded 3d with a service in a cloud. But I will see if that could be automated.

**Comment by MaticDiba** [comment]
Hi,
I tested my model on your demo website. It seems like it works.

I guess now, I will just have to figure it out how to use las or asc files and convert them into tiles with grass library. 

I must say, CaveView really is an awesome library! Thumbs up!!!

**Comment by aardgoose** [comment]
Hi,

I have put up some documentation for tile creation [here](https://aardgoose.github.io/CVTiles/). 

Hope this is useful and let me known about any errors in the docs.

Cheers

**Comment by MaticDiba** [comment]
I read the documentation and it's clear what I have to do. I think there is a typo in "tilecreation.html", where you mention "ESPG:3875" CRS, but I think it's 3875. But it's not a big deal. 

The other thing is, I tried to use Cesium (I signed up and used my own apikey) and it seems to work with provided examples of surveys. But when I tried to use my own survey, left menu bar is not rendered. If I drag it to https://aardgoose.github.io/CaveView.js/fulldemo.html it works (menu bar is displayed). This is the [project](https://we.tl/t-6ZCn7q7XbU) I was playing with.

Oh, and the link to cesiumtiles.html is dead.

**Comment by aardgoose** [comment]
Hi,

when the menu bar doesn't come up, its probably an error which isn't handled well. If you can point me to a demo of this I can take a look.  The first place is the javascript console in the browser, although if you use the minimized builds the error information will be rather cryptic.
